src/scenic/core/scenarios.py

```diff
@@ -1,518 +1,595 @@
+        self.terminateSimulationConditions = tuple(dynamicScenario._terminateSimulationConditions)
+        self.initialRequirements = self.requirements
+        assert all(req.constrainsSampling for req in self.initialRequirements)
+        self.recordedExprs = tuple(dynamicScenario._recordedExprs)
+        self.recordedInitialExprs = tuple(dynamicScenario._recordedInitialExprs)
+        self.recordedFinalExprs = tuple(dynamicScenario._recordedFinalExprs)
+        # dependencies must use fixed order for reproducibility
+        paramDeps = tuple(p for p in self.params.values() if isinstance(p, Samplable))
+        behaviorDeps = []
+        for namespace in self.behaviorNamespaces.values():
+            for value in namespace.values():
+                if isinstance(value, Samplable):
+                    behaviorDeps.append(value)
+        self.dependencies = self._instances + paramDeps + tuple(requirementDeps) + tuple(behaviorDeps)
+
+        self.validate()
+
+    def containerOfObject(self, obj):
+        if hasattr(obj, 'regionContainedIn') and obj.regionContainedIn is not None:
+            return obj.regionContainedIn
+        else:
+            return convertToFootprint(self.workspace.region)
+
+    def validate(self):
+        """Make some simple static checks for inconsistent built-in requirements.
+
+        :meta private:
+        """
+        objects = self.objects
+        staticVisibility = self.egoObject and not needsSampling(self.egoObject.visibleRegion)
+        staticBounds = [obj._hasStaticBounds for obj in objects]
+        for i in range(len(objects)):
+            oi = objects[i]
+            container = self.containerOfObject(oi)
+            # Trivial case where container is empty
+            if isinstance(container, EmptyRegion):
+                raise InvalidScenarioError(f'Container region of {oi} is empty')
+            # skip objects with unknown positions or bounding boxes
+            if not staticBounds[i]:
+                continue
+            # Require object to be contained in the workspace/valid region
+            if not needsSampling(container) and not container.containsObject(oi):
+                raise InvalidScenarioError(f'Object at {oi.position} does not fit in container')
+            # Require object to be visible from the ego object
+            if staticVisibility and oi.requireVisible is True and oi is not self.egoObject:
+                if not self.egoObject.canSee(oi):
+                    raise InvalidScenarioError(f'Object at {oi.position} is not visible from ego')
+            if not oi.allowCollisions:
+                # Require object to not intersect another object
+                for j in range(i):
+                    oj = objects[j]
+                    if oj.allowCollisions or not staticBounds[j]:
+                        continue
+                    if oi.intersects(oj):
+                        oi_name = oi.name if hasattr(oi, 'name') else "Object"
```

Comment:
More idiomatic:
```suggestion
                        oi_name = getattr(oi, 'name', 'Object')
```
But I think just using `{oi}` below is even better since objects have an informative `__str__`.